tensorflow/src/main/scala/com/spotify/featran/tensorflow/ExampleTransformer.scala

```diff
@@ -0,0 +1,186 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.featran.tensorflow
+
+import com.spotify.featran.{CollectionType, FeatureBuilder, JsonSerializable}
+import com.spotify.featran.transformers._
+import org.tensorflow.example.{Example, Feature}
+import shapeless.datatype.tensorflow.TensorFlowType
+
+import scala.reflect.ClassTag
+
+/**
+ * ExampleTransformer takes a TFExample of non-transformed values output by Featran
+ * and applies a settings file to transform the Features into the required output object.
+ *
+ * The main benefit on this approach is that when using this class there does not have to
+ * be a JVM dependency on the original scala object or the original spec that were used to
+ * create the TFExample and the Settings file.
+ *
+ *
+ */
+object ExampleTransformer {
+  import TensorFlowType._
+
+  def toFeature(name: String, ex: Example): Option[Feature] = {
+    val fm = ex.getFeatures.getFeatureMap
+    if (fm.containsKey(name)) {
+      Some(fm.get(name))
+    } else {
+      None
+    }
+  }
+
+  def getDouble(name: String): Example => Option[Double] =
+    (ex: Example) => toFeature(name, ex).flatMap(v => toDoubles(v).headOption)
+
+  def getMdlRecord(name: String): Example => Option[MDLRecord[String]] =
+    (ex: Example) => {
+      for {
+        labelFeature <- toFeature(name + "_label", ex)
+        label <- toStrings(labelFeature).headOption
+        valueFeature <- toFeature(name + "_value", ex)
+        value <- toDoubles(valueFeature).headOption
+      } yield MDLRecord(label, value)
+    }
+
+  def getWeightedLabel(name: String): Example => Option[List[WeightedLabel]] =
+    (ex: Example) => {
+      val labels = for {
+        keyFeature <- toFeature(name + "_key", ex).toList
+        key <- toStrings(keyFeature)
+        valueFeature <- toFeature(name + "_value", ex).toList
+        value <- toDoubles(valueFeature)
+      } yield WeightedLabel(key, value)
+      if (labels.isEmpty) None else Some(labels)
+    }
+
+  def getDoubles(name: String): Example => Option[Seq[Double]] =
+    (ex: Example) => toFeature(name, ex).map(v => toDoubles(v))
+
+  def getDoubleArray(name: String): Example => Option[Array[Double]] =
+    (ex: Example) => toFeature(name, ex).map(v => toDoubles(v).toArray)
+
+  def getString(name: String): Example => Option[String] =
+    (ex: Example) => toFeature(name, ex).flatMap(v => toStrings(v).headOption)
+
+  def getStrings(name: String): Example => Option[Seq[String]] =
+    (ex: Example) => toFeature(name, ex).map(v => toStrings(v))
+
+  def apply[M[_]: CollectionType](settings: M[String]): ExampleTransformer[M] =
+    new ExampleTransformer[M](settings)
+}
+
+class ExampleTransformer[M[_]: CollectionType](settings: M[String]) extends Serializable {
```

Comment:
I'm not sure if I like the API, specifically `ExampleTransformer` suggests that it might be an implementation of a `Transformer` but it's not. It's more like some form of `Extractor` - kinda like `FeatureExtractor`, but specifically for `Example`s.

Reply:
@ravwojdyla yeah kinda leanning towards `ExampleExtractor` as well

Reply:
ExampleExtractor?  Lots of extracts/converts/transform names floating around that can be confusing.

Reply:
@richwhitjr yeah naming right? 😄